extension/src/Package.ts

```diff
@@ -286,6 +287,20 @@ function uriJoin(directory: vscode.Uri, file: string) {
 
 function findVsixFile(manifest: PackageManifest) {
     if (manifest.files) {
+        if (manifest.osSpecific) {
+            const os = require("os");
+            const osEntry = manifest.osSpecific[os.platform()];
+
+            if (osEntry) {
+                const osSpecificFile = manifest.files.find(e => (e === osEntry && typeof e === 'string' && e.endsWith('.vsix')));
+
+                if (osSpecificFile) {
+                    return osSpecificFile;
+                }
+            }
+            return null;
```

Comment:
Should we support some sort of "default" value to use in case you're using an OS that doesn't match any listed ones? It seems like that could be useful for cases where your extension can run with reduced features when it doesn't have the native dependencies.

Reply:
The use cases I have encountered so far are OS specific only, but it could be useful though. What about adding an optional "default" OS?

Reply:
That seems reasonable to me.

Also, in the case where there is no "default" value, it would be nice if the [error message](https://github.com/joelspadin-garmin/vscode-private-extension-manager/blob/1f4471728c80fa7922fc06aad829d8a935dbab36/extension/src/Package.ts#L199) shown to the user  specifically mentions that the extension does not support their OS. I'm not sure what the cleanest way to fit that into the existing system would be though. A couple ideas:

1. Instead of returning `null` when we don't find a package, throw one exception for unsupported OS and a different exception for no .vsix file. Catch those in `Package`'s constructor, set `this.vsixFile = null`, and set some error state so we can report why the file is missing.
2. Instead of returning `null` when we don't find a package, return an error state enum. Then `this.vsixFile` is always either a valid file or a reason why we couldn't find a file, but then the other code that relies on it being truthy when valid would need to be changed.

Reply:
I'll have a look at it, thanks!

Reply:
`default` and error handling added.